docs/rfcs/rfc-040-relay-first-api.md

```diff
@@ -0,0 +1,1347 @@
+    AND: [MovieActorsConnectionWhere!]
+    OR: [MovieActorsConnectionWhere!]
+    NOT: MovieActorsConnectionWhere
+    edges: MovieActorsWhereFilters
+    aggregation: MovieActorsAggregationEdgeWhere
+}
+
+input MovieActorsWhereFilters {
+    AND: [MovieActorsWhereFilters!]
+    OR: [MovieActorsWhereFilters!]
+    NOT: MovieActorsWhereFilters
+    all: MovieActorsEdgeWhere
+    none: MovieActorsEdgeWhere
+    single: MovieActorsEdgeWhere
+    some: MovieActorsEdgeWhere
+}
+
+input MovieDirectorConnectionWhere {
+    AND: [MovieDirectorConnectionWhere!]
+    OR: [MovieDirectorConnectionWhere!]
+    NOT: MovieDirectorConnectionWhere
+    edges: MovieDirectorEdgeWhere
+}
+
+input PersonMoviesConnectionWhere {
+    AND: [PersonMoviesConnectionWhere!]
+    OR: [PersonMoviesConnectionWhere!]
+    NOT: PersonMoviesConnectionWhere
+    edges: PersonMoviesWhereFilters
+    aggregation: PersonMoviesAggregationEdgeWhere
+}
+
+input PersonMoviesWhereFilters {
+    AND: [PersonMoviesWhereFilters!]
+    OR: [PersonMoviesWhereFilters!]
+    NOT: PersonMoviesWhereFilters
+    all: PersonMoviesEdgeWhere
+    none: PersonMoviesEdgeWhere
+    single: PersonMoviesEdgeWhere
+    some: PersonMoviesEdgeWhere
+}
+
+input PersonDirectedConnectionWhere {
+    AND: [PersonDirectedConnectionWhere!]
+    OR: [PersonDirectedConnectionWhere!]
+    NOT: PersonDirectedConnectionWhere
+    edges: PersonDirectedEdgeWhere
+}
+
+## Relationship Edge Where
+
+input MovieActorsEdgeWhere {
+    AND: [MovieActorsEdgeWhere!]
+    OR: [MovieActorsEdgeWhere!]
+    NOT: MovieActorsEdgeWhere
+    node: PersonNodeWhere
+    fields: ActedInWhere
+}
+
+input MovieDirectorEdgeWhere {
+    AND: [MovieDirectorEdgeWhere!]
+    OR: [MovieDirectorEdgeWhere!]
+    NOT: MovieDirectorEdgeWhere
+    node: PersonNodeWhere
+    fields: ActedInWhere
+}
+
+input PersonMoviesEdgeWhere {
+    AND: [PersonMoviesEdgeWhere!]
+    OR: [PersonMoviesEdgeWhere!]
+    NOT: PersonMoviesEdgeWhere
+    node: MovieNodeWhere
+    fields: ActedInWhere
+}
+
+input PersonDirectedEdgeWhere {
+    AND: [PersonDirectedEdgeWhere!]
+    OR: [PersonDirectedEdgeWhere!]
+    NOT: PersonDirectedEdgeWhere
+    node: MovieNodeWhere
+}
+
+## Relationship fields where
+
+input ActedInWhere {
+    OR: [ActedInWhere!]
+    AND: [ActedInWhere!]
+    NOT: ActedInWhere
+    year: IntWhere
+}
+
+## Aggregation filters
+
+input MovieActorsAggregationEdgeWhere {
+    AND: [MovieActorsAggregationEdgeWhere!]
+    OR: [MovieActorsAggregationEdgeWhere!]
+    NOT: MovieActorsAggregationEdgeWhere
+    nodes: PeopleAggregationWhere
+    fields: ActedInAggregationEdgeWhere
+}
+
+input PersonMoviesAggregationEdgeWhere {
+    AND: [PersonMoviesAggregationEdgeWhere!]
+    OR: [PersonMoviesAggregationEdgeWhere!]
+    NOT: PersonMoviesAggregationEdgeWhere
+    nodes: MoviesAggregationWhere
+    edges: ActedInAggregationEdgeWhere
+}
+
+input ActedInAggregationEdgeWhere {
+    fields: ActedInAggregationWhere
+    count: IntWhere
```

Comment:
I guess that if we are not addressing the support of multiple edges between two nodes in this phase/RFC, it may be better removes all the related references, like counting the edges. Thought?

Reply:
I think is worth adding optional capabilities for the future, I'll add a comment however to make sure it is clear

Reply:
Yeah, you are probably right here